crates/walking_inference/src/inputs.rs

```diff
@@ -0,0 +1,174 @@
+    parameters::{MotorCommandParameters, RLWalkingParameters},
+};
+
+#[derive(
+    Debug, Default, Clone, Serialize, Deserialize, PathSerialize, PathDeserialize, PathIntrospect,
+)]
+pub struct WalkingInferenceInputs {
+    pub gravity: Vector3<Robot>,
+    pub angular_velocity: Vector3<Robot>,
+    pub linear_velocity_command: Vector2<Ground>,
+    pub angular_velocity_command: f32,
+    pub gait_process: nalgebra::Vector2<f32>,
+    pub joint_position_differences: [f32; 12],
+    pub joint_velocities: [f32; 12],
+    pub last_target_joint_positions: [f32; 12],
+}
+
+impl WalkingInferenceInputs {
+    #[allow(clippy::too_many_arguments)]
+    pub fn try_new(
+        now: SystemTime,
+        roll_pitch_yaw: Vector3<Robot>,
+        angular_velocity: Vector3<Robot>,
+        motion_command: &MotionCommand,
+        current_serial_joints: Joints<MotorState>,
+        last_target_joint_positions: Joints,
+        last_linear_velocity_command: Vector2<Ground>,
+        last_angular_velocity_command: f32,
+        walking_parameters: RLWalkingParameters,
+        motor_command_parameters: MotorCommandParameters,
+    ) -> Result<Self> {
+        let policy_interval = walking_parameters.control.dt * walking_parameters.control.decimation;
+
+        let (linear_velocity_command, angular_velocity_command) = match motion_command {
+            MotionCommand::WalkWithVelocity {
+                velocity,
+                angular_velocity,
+                ..
+            } => {
+                let linear_velocity_command_difference = velocity - last_linear_velocity_command;
+                let angular_velocity_command_difference =
+                    angular_velocity - last_angular_velocity_command;
+                (
+                    last_linear_velocity_command
+                        + vector!(
+                            linear_velocity_command_difference
+                                .x()
+                                .clamp(-policy_interval, policy_interval,),
+                            linear_velocity_command_difference
+                                .y()
+                                .clamp(-policy_interval, policy_interval,)
+                        ),
+                    last_angular_velocity_command
+                        + angular_velocity_command_difference
+                            .clamp(-policy_interval, policy_interval),
+                )
+            }
+            _ => todo!(),
+        };
+
+        let gait_frequency = if linear_velocity_command.norm() < 1e-5 {
+            0.0
+        } else {
+            walking_parameters.gait_frequency
+        };
+        let gait_progress = (gait_frequency * now.duration_since(UNIX_EPOCH)?.as_secs_f32()) % 1.0;
+
+        let is_walking = gait_frequency > 1.0e-8;
+        let gait_process = if is_walking {
+            nalgebra::Rotation2::new(2.0 * PI * gait_progress) * nalgebra::Vector2::x()
+        } else {
+            Default::default()
```

Comment:
Through experimentation I found that this branch is what causes the feet drifting apart when not walking.
I'd suggest simply removing the whole if clause and always generating a rotated unit vector which may happen to be static if the frequency is 0.